internal/app/util.go

```diff
@@ -233,6 +235,32 @@ func isGTIDLessOrEqual(slaveGtidSet, masterGtidSet gtids.GTIDSet) bool {
 	return masterGtidSet.Contain(slaveGtidSet) || masterGtidSet.Equal(slaveGtidSet)
 }
 
+func isSplitBrained(slaveGtidSet, masterGtidSet *gomysql.MysqlGTIDSet, masterUUID uuid.UUID) bool {
+	for _, masterSet := range masterGtidSet.Sets {
+		slaveSet, ok := slaveGtidSet.Sets[masterSet.SID.String()]
```

Comment:
In this case we will treat replica splitbrained just after new master appears and has no commited transactions.

May be we should only cycle through slaveGtidSet.Sets and compare them with master?

Reply:
this is not-splitbrain, I guess (trx with gtid uuid1 may be huge):
```
  master  |  replica 
----------+----------
uuid1:123 |
uuid2:123 | uuid2:123
uuid3:123 | uuid3:123
```
only this:
```
  master  |  replica 
----------+----------
uuid2:123 | uuid2:123
uuid3:123 | uuid3:123
          | uuid4:123
```

Reply:
Yes, I agree. This is implemented through conditions on 241 lines